src/features/search/search-slice.ts

```diff
@@ -19,8 +24,15 @@ export const searchSlice = createSlice({
     setSearchTerm: (state, action: PayloadAction<string>) => {
       state.searchTerm = action.payload;
     },
+    setTempSearchTerm: (state, action: PayloadAction<string>) => {
```

Comment:
I feel like either TempSearchTerm and SearchPreviewScrollLeft need better names or they need comments somewhere, as it was difficult through this PR to understand what they did